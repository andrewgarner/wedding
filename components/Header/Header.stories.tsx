import { Meta, Story } from "@storybook/react";
import React from "react";
import { Header } from "./Header";

export default {
  title: "Layout/Header",
  component: Header,
} as Meta;

const Template: Story = (args) => <Header {...args} />;

export const Default = Template.bind({});
