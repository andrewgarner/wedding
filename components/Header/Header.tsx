import Link from "next/link";
import React, { FunctionComponent } from "react";
import { Text, Typography } from "../Text";

enum HeaderDisplay {
  Visible = "Visible",
  Invisible = "Invisible",
  Hidden = "Hidden",
}

const headerDisplayClassNames: Record<HeaderDisplay, string> = {
  [HeaderDisplay.Visible]: "visible",
  [HeaderDisplay.Invisible]: "invisible h-0",
  [HeaderDisplay.Hidden]: "hidden",
};

interface HeaderProps {
  display?: HeaderDisplay;
}

const Header: FunctionComponent<HeaderProps> = ({
  display = HeaderDisplay.Visible,
}) => (
  <header className={headerDisplayClassNames[display]}>
    <Text tag="h1" typography={Typography.HeaderXLarge}>
      <Link href="/">
        <a>Andrew and Hannah</a>
      </Link>
    </Text>
  </header>
);

export { Header, HeaderDisplay };
export type { HeaderProps };
