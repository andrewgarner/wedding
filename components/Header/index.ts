export { Header, HeaderDisplay } from "./Header";
