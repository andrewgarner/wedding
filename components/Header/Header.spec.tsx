import { render } from "@testing-library/react";
import React from "react";
import { Header } from "./Header";

describe("The header component", () => {
  it("renders correctly", () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
