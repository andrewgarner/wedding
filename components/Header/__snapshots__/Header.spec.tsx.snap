// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The header component renders correctly 1`] = `
<div>
  <header
    class="visible"
  >
    <h1
      class="space-y-32 leading-loose  first-of-type:mt-0 mt-32 mb-4 mx-auto font-sans text-3xl lg:text-4xl xl:text-5xl font-normal uppercase"
    >
      <a
        href="/"
      >
        Andrew and Hannah
      </a>
    </h1>
  </header>
</div>
`;
